workflow_plandefinition/models/workflow_plan_definition_action.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import fields, models, api, exceptions
+
+
+class PlanDefinitionAction(models.Model):
+    _name = 'workflow.plan.definition.action'
+    _description = 'Medical Plan Definition Action'
+    _inherit = 'mail.thread'
+    _parent_name = 'parent_id'
+    _parent_store = True
+    _parent_order = 'name'
+    _order = 'parent_left'
+    _rec_name = 'complete_name'
+    _order = 'id asc'
+
+    @api.one
+    @api.depends('name', 'parent_id')
+    def _compute_complete_name(self):
+        """ Forms complete name of action from parent to child action. """
+        name = self.name
+        current = self
+        while current.parent_id:
+            current = current.parent_id
+            name = '%s/%s' % (current.name, name)
+        self.complete_name = name
+
+    @api.multi
+    @api.depends('parent_id', 'direct_plan_definition_id')
+    def _compute_plan_definition_id(self):
+        for rec in self:
+            rec.plan_definition_id = rec.direct_plan_definition_id
+            if rec.parent_id:
+                rec.plan_definition_id = rec.parent_id.plan_definition_id
+
+    name = fields.Char(
```

Comment:
Move field definitions to top

Reply:
What do you mean? To move all the compute methods at the end of the file?

Reply:
Oops sorry I should have linked [contribution docs section](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#models). Yup, exactly:

In a Model attribute order should be
1. Private attributes (_name, _description, _inherit, ...)
1. Default method and _default_get
1. Fields declarations
1. Compute and search methods in the same order than field declaration
1. Constrains methods (@api.constrains) and onchange methods (@api.onchange)
1. CRUD methods (ORM overrides)
1. Action methods
1. And finally, other business methods.

Reply:
Oke, I didn't know about these guidelines. Fixed!